hydrolib/core/basemodel.py

```diff
@@ -530,22 +593,30 @@ def __new__(cls, filepath: Optional[Path] = None, *args, **kwargs):
             else:
                 return super().__new__(cls)
 
-    def __init__(self, filepath: Optional[Path] = None, *args, **kwargs):
+    def __init__(
+        self, filepath: Optional[Path] = None, resolve_casing=False, *args, **kwargs
```

Comment:
Just wondering, should we use `Optional[bool] = None` instead? That's the way it is used for the `filepath` argument. If not, why not?

Reply:
Yes!

Reply:
I will make it `resolve_casing: bool = False`, because it is optional but it is not `Optional`.
`Optional[bool]` means that it is a union between `bool` and `None`. 
It is _optional_ for a user to pass because we have provided a default value.